### PR TITLE
Remove usage of obsolete property $_isScopePrivate

### LINKED ISSUE #30506: More than core 80 blocks use _isScopePrivate in spite of being officially described as _not working properly_
### Preconditions (*)

Magento 2.3.5
Magento 2.4-develop 

### Steps to reproduce (*)

Search for the string ` $this->_isScopePrivate = true` inside **vendor/magento**.

### Expected result (*)

Since using the $_isScopePrivate property in blocks is obsolete and will not work properly, it should not be used by any block provided by Magento 2.

See: https://devdocs.magento.com/guides/v2.3/extension-dev-guide/cache/page-caching/private-content.html#config-cache-priv-how-block

### Actual result (*)

More than 80 (eighty) blocks use that property, potentially leading to unexpected results, because `Do not use the $_isScopePrivate property in your blocks. This property is obsolete and **will not work properly**`.

### Thoughts

Fixing this issue is nearly impossible for developers. It mean overriding every blocks that use that property. This needs to be fixed upstream. Or if the property works, the documentation needs to be updated.

As a matter of fact, it created some strange cache issues on blocks inheriting from **\Magento\Checkout\Block\Onepage\Link** where the block would first appear as expected then be replaced a few tenth of seconds later by a previsously cached version, probably grabbed from the local storage.

As long as this issue has not been fixed, we can't trust any of the 80+ concerned blocks: instanciation one of them or inheriting from one of them can lead to unexpected results.

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [x] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### PR DESCRIPTION
**Pull request by mrtuvn**
### Description (*)

In old codebase developers still use this property in someplace. We should clean that to follow best practices
https://devdocs.magento.com/guides/v2.4/extension-dev-guide/cache/page-caching/private-content.html#config-cache-priv-how-block
Search results return around 59 blocks still have this property. It's quite lot and can make break change if we refactor this

Just wonder if blocks created not from layout and created via other place (Controllers, Template view), How we can manager cache for that after remove isScopePrivate property

### Fixed Issues (if relevant)

1. Fixes magento/magento2#30506

### Manual testing scenarios (*)

1. ...
2. ...

### Questions or comments

CC: @ihor-sviziev @ericmorand @marvinhinz @orlangur @sidolov 
Feel free to give me any comments

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by ihor-sviziev** [comment]
TBH I don't think it's a bad practice to use _isScopePrivate. It would be good to understand what issues it's causing.
The cacheable=false causing the entire page to be not cached at all.

**Comment by mrtuvn** [comment]
We should update more in docs related with this. Maybe add some real world use case for example. Seem  current it's not clearly at my point of view

**Comment by ihor-sviziev** [comment]
It seems like a message about not using _isScopePrivat was added a long time ago, in 2016 https://github.com/magento/devdocs/commit/a75a7a91892ecf115a52f8380580530af2719358#diff-21ef8286af3f0fb99ca6976a2c50f78693f2fe7e46b6394635ac8112792087adR62-R64.

@slavvka @sidolov, could you explain why this approach was deprecated and should work incorrectly but still used in the core?

**Comment by sivaschenko** [comment]
@ihor-sviziev I believe the reason to walk away from `$_isScopePrivate` is to handle and invalidate private data more precisely that is possible with customer data sections.

The property was not yet deprecated by core teams due to low priority, internal ticket: https://jira.corp.magento.com/browse/MAGETWO-56701

**Comment by sivaschenko** [comment]
This pull request not fixing but is related to the issue https://github.com/magento/magento2/issues/30506 (just to create a link)

**Comment by sivaschenko** [comment]
I'd probably try to deprecate the property in 2.4 and remove in 2.5, @gabrieldagama @sidolov what do you think?

**Comment by mrtuvn** [comment]
Should i re-target branch and update issue link ?

**Comment by sivaschenko** [comment]
@mrtuvn considering the tumb-ups, yes, please

Thanks! :)

File: app/code/Magento/Multishipping/Block/Checkout/Address/Select.php [hunk 1]
```diff
@@ -28,11 +28,6 @@ class Select extends \Magento\Multishipping\Block\Checkout\AbstractMultishipping
      */
     protected $addressMapper;
 
-    /**
-     * @var bool
-     */
-    protected $_isScopePrivate = true;
```
Comment (sivaschenko) [inline app/code/Magento/Multishipping/Block/Checkout/Address/Select.php]: While it should be ok to remove the redeclarations of the inherited property with the same value, I don't think we should remove the `$_isScopePrivate = true` in this pull request (that should be done in 2.5 as part as deprecated property removal preparation)

**Comment by sivaschenko** [comment]
Prioritized as P2 as missing the deprecated tag on the property prevents any further progress in moving to best practices in this direction

File: app/code/Magento/Multishipping/Block/Checkout/Address/Select.php (hunk 1 above)
Comment (mrtuvn) [reply app/code/Magento/Multishipping/Block/Checkout/Address/Select.php]: should i need revert all that ? only keep deprecate notes

Comment (sivaschenko) [reply app/code/Magento/Multishipping/Block/Checkout/Address/Select.php]: This is the only file where the removal should be reverted

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-8668](https://jira.corp.magento.com/browse/ENGCOM-8668) has been created to process this Pull Request

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA Passed

### Manual testing scenario
1. Open Magento project in PHP Storm, for example
2. Go to /app/code/Magento
3. Search for text `$this->_isScopePrivate = true`

**AFTER** applying changes provided in this PR, all the `$this->_isScopePrivate = true` are marked as **Deprecated**
see an example
is_deprecated

**Comment by mrtuvn** [comment]
Only rebase code nothing else change

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA Passed
The result is the same as in the report above https://github.com/magento/magento2/pull/31754#issuecomment-763564752

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B

**Pull request merged**